notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1003 @@
+        2.25    2.89   -2.85   -6.14    0.72] | cost: 64.05
+    theta: [-218.57    0.93    1.7     0.85  -11.66   -7.67   -5.98    7.36   13.63
+        2.27    2.86   -2.89   -6.12    0.69] | cost: 64.05
+    theta: [-218.75    0.93    1.69    0.85  -11.61   -7.66   -5.98    7.39   13.58
+        2.26    2.81   -2.94   -6.12    0.65] | cost: 64.04
+    theta: [-218.83    0.93    1.69    0.85  -11.58   -7.64   -5.98    7.44   13.52
+        2.21    2.72   -3.02   -6.16    0.57] | cost: 64.04
+    theta: [-218.74    0.93    1.69    0.85  -11.62   -7.63   -5.98    7.48   13.46
+        2.13    2.61   -3.08   -6.24    0.5 ] | cost: 64.04
+    theta: [-218.58    0.93    1.69    0.85  -11.67   -7.62   -5.98    7.48   13.43
+        2.07    2.55   -3.11   -6.3     0.46] | cost: 64.04
+    theta: [-218.51    0.93    1.7     0.85  -11.69   -7.62   -5.97    7.46   13.43
+        2.05    2.55   -3.11   -6.33    0.44] | cost: 64.04
+    theta: [-218.49    0.93    1.7     0.85  -11.7    -7.62   -5.96    7.44   13.43
+        2.04    2.55   -3.11   -6.34    0.42] | cost: 64.04
+    theta: [-218.51    0.93    1.7     0.85  -11.69   -7.62   -5.96    7.42   13.43
+        2.04    2.56   -3.11   -6.36    0.4 ] | cost: 64.04
+    theta: [-218.54    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.42
+        2.04    2.57   -3.12   -6.36    0.39] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    theta: [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
+        2.04    2.57   -3.12   -6.36    0.38] | cost: 64.04
+    
+
+Looks like gradient descent converged to those theta values! Thus, our linear model is:
+
+$y = -218.56 + 0.93x_1 + ... -6.36x_{13} + 0.38x_{14}$
+
+Let's compare this equation that we obtained to the one we would get if we had used `sklearn`'s LinearRegression model instead. The LinearRegression model automatically includes a column of all 1's in our data, so we can get rid of our column of 1's when we pass `X` into the `fit` function.
+
+
+```python
+len(donkeys_c.columns.values)
+```
+
+
+
+
+    15
+
+
+
+
+```python
```

Comment:
Try this in the codeblock instead

```python
model = LinearRegression(fit_intercept=False) # We already accounted for it with the bias column
model.fit(X[:, :14], y)
print("Coefficients", model.coef_)
```

Reply:
The result will be 
`Coefficients [-218.56    0.93    1.7     0.85  -11.68   -7.61   -5.96    7.41   13.41
    2.04    2.57   -3.12   -6.36    0.38]`

Reply:
wow you are the coolest! thanks for finding this out for me <3